Issues/82/Support for .ico

AMorgaut opened the pull request:
Implementation proposal for issue #82

netroy commented:
@AMorgaut looks good.. thanks :)
regarding the spec images: are they free to use? if not, we should create our own (or get something released under creative commons)

AMorgaut commented:
I thought it was but looking more closely it looks it might not be an open enough license

> http://www.iconarchive.com/show/pretty-office-4-icons-by-custom-icon-design/ICO-icon.html

It is said "Free for non-commercial use"
It might be ok if you add a `.npmignore` file that filter the `specs` folder (usually a good practice IMHO) as it would only be used in dev mode on a non-commercial project. 

But still I thinks its better to fix this as I'm sure we'll find a more open one and as we really don't care what the icon looks like

The other ico files are editions done via GIMP of this same "multi" icon to handle the different specific ico variants

Regarding the CUR file, it is _Released under the **Release to Public Domain** license._

> http://www.rw-designer.com/cursor-detail/13156

This license is described this way: 

_You are free: To use this work for any legal purpose._

> http://www.rw-designer.com/licenses

AMorgaut commented:
As the package.json specified that the nodejs engine had to be ">=0.10.0":

- I first added the "0.10" & "0.12" targets to the travis config
- I installed nodejs "0.10.48" and tested it locally on my machine
- I replaced the non-supported "const" statements by "var" ones in my code

Unfortunately there was still a buffer issue in the "webp" image size reader for the 0.10 target, So I rolled back to a 0.12 minimal target and then everything went fine, as you can see, even on travis

Release Note: 

- "package.json" switch minimal node version from 0.10 to 0.12
- "travis.yml" includes the 0.12 target & switch the version 7 by the new 8 LTS version (as it already jumped from 4 to 6)
- I re-managed a little my ico assets to better highlight the 256 pixels resolution special case (shown as size zero in .ico binary files)
- I included a license file in the ico folder to better comply with the creative common license
- The test file now filter any ".md" (potential license) file in the "specs/image" folders
- I added a ".npmignore" file that filter the "specs" folder when publishing to npm so: 
  - the npm published version is lighter and faster to install
  - it doesn't contain any file potentially requiring additional license

netroy commented on .travis.yml:
```diff
@@ -4,9 +4,10 @@ git:
 language: node_js
 cache: yarn
 node_js:
+  - 0.12
```
0.12 is already end-of-life, we should remove this.

AMorgaut replied:
make sense ok

netroy commented on .npmignore:
```diff
@@ -0,0 +1,9 @@
+.DS_Store
```
[we already use a whitelist](https://github.com/image-size/image-size/blob/master/package.json#L6) .. this file can be removed.

AMorgaut replied:
cool ok

netroy commented on package.json:
```diff
@@ -8,7 +8,7 @@
   ],
   "engines": {
-    "node": ">=0.10.0"
+    "node": ">=0.12.0"
```
I'd prefer if we change this to 4.x instead.

netroy commented:
published as `image-size@0.6.0`
